modules/apps/transfer/types/encoding.go

```diff
@@ -0,0 +1,42 @@
+package types
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+// UnmarshalJSON implements the Unmarshaller interface for FungibleTokenPacketDataV2.
+func (ftpd *FungibleTokenPacketDataV2) UnmarshalJSON(bz []byte) error {
+	// Recursion protection.
+	type ftpdAlias FungibleTokenPacketDataV2
```

Comment:
is there a way to add a test which breaks things if this is removed/refactored?

Reply:
Yeah, since the code is pretty much the same for both, this could be a good opportunity to use generics :)

Reply:
I think I would lean on the side of making things simpler, it's not a lot to copy paste, and the behaviour will what is expected.